redi-connect-front/src/serviceWorker.js

```diff
@@ -0,0 +1,185 @@
+    if (navigator.onLine) {
+      config.onOnline()
+    } else {
+      config.onOffline()
+    }
+  }
+}
+
+export function register (config) {
+  if (process.env.NODE_ENV === 'production' && 'serviceWorker' in navigator) {
+    // The URL constructor is available in all browsers that support SW.
+    const publicUrl = new URL(process.env.PUBLIC_URL, window.location.href)
+    if (publicUrl.origin !== window.location.origin) {
+      // Our service worker won't work if PUBLIC_URL is on a different origin
+      // from what our page is served on. This might happen if a CDN is used to
+      // serve assets; see https://github.com/facebook/create-react-app/issues/2374
+      console.log(publicUrl.origin, window.location.origin)
+      return
+    }
+
+    window.addEventListener('load', () => {
+      const swUrl = `${process.env.PUBLIC_URL}/service-worker.js`
+
+      if (isLocalhost) {
+        // This is running on localhost. Let's check if a service worker still exists or not.
+        checkValidServiceWorker(swUrl, config)
+
+        // Add some additional logging to localhost, pointing developers to the
+        // service worker/PWA documentation.
+        navigator.serviceWorker.ready.then(() => {
+          // This web app works offline!
+          console.log(
+            'This web app is being served cache-first by a service ' +
+              'worker. To learn more, visit http://bit.ly/CRA-PWA'
+          )
+        })
+      } else {
+        // Is not localhost. Just register service worker
+        registerValidSW(swUrl, config)
+      }
+    })
+
+    window.addEventListener('offline', () => handleNetworkChange(config))
+    window.addEventListener('online', () => handleNetworkChange(config))
+
+    // let deferredPrompt
+
+    // window.addEventListener('beforeinstallprompt', (e) => {
+    //   // Prevent Chrome 67 and earlier from automatically showing the prompt
+    //   e.preventDefault()
+    //   // Stash the event so it can be triggered later.
+    //   deferredPrompt = e
+    //   // Update UI notify the user they can add to home screen
+    //   btnAdd.style.display = 'block'
+    // })
+    // btnAdd.addEventListener('click', (e) => {
+    //   // hide our user interface that shows our A2HS button
+    //   btnAdd.style.display = 'none'
+    //   // Show the prompt
+    //   deferredPrompt.prompt()
+    //   // Wait for the user to respond to the prompt
+    //   deferredPrompt.userChoice
+    //     .then((choiceResult) => {
+    //       if (choiceResult.outcome === 'accepted') {
+    //         console.log('User accepted the A2HS prompt')
+    //       } else {
+    //         console.log('User dismissed the A2HS prompt')
+    //       }
+    //       deferredPrompt = null
+    //     })
+    // })
+    // window.addEventListener('appinstalled', (evt) => {
+    //   app.logEvent('a2hs', 'installed')
+    // })
+  } else {
+    console.log('browser does not support sw')
```

Comment:
Does this will be logging also in production code?

Reply:
yes, I don't think create-react-app's webpack config removes console logs in production